Reapply setup vespalog in jdisc core

gjoranv opened the pull request.

gjoranv committed:
Reapply setup of vespalog in jdisc_core (#22509)

Revert "Merge pull request #22515 from vespa-engine/revert-22509-setup-vespalog-in-jdisc_core"

gjoranv committed:
Remove duplicate call to InitializeLog.init()

hakonhall commented on container-disc/src/main/java/com/yahoo/container/jdisc/ConfiguredApplication.java:
```diff
@@ -42,7 +42,7 @@
 import com.yahoo.jrt.slobrok.api.Register;
 import com.yahoo.jrt.slobrok.api.SlobrokList;
-import com.yahoo.log.LogSetup;
+import com.yahoo.log.impl.InitializeLog;
```
Stray import.

hakonhall commented on vespalog/src/main/java/com/yahoo/log/impl/InitializeLog.java:
```diff
@@ -0,0 +1,21 @@
+
+    public static void init(String programName) {
+        LogSetup.initVespaLogging(programName);
+        log.log(Level.INFO, "Initialized logging for " + programName);
```
When sifting through large amounts of logs it is often useful to know when a process started.  That was the intention of the `Starting jdisc at version 7.581.34` message.  If the initialization of the logging is moved so should that message (and this message is redundant).

gjoranv replied:
This is just a draft, not ready for review yet :)

gjoranv replied:
The log message you mention could be moved to ApplicationLoader, but without outputting the version which is not available for jdisc_core. Making it available is outside the scope of this PR.

hakonhall replied:
Not sure what's best.

gjoranv committed:
Set up vespalog in StandaloneMain static initializer

- Remove unused InitializeLog

gjoranv committed:
Do not install OsgiLogManager

- It mutes vespa.log and does not seem to do anything useful
  for current usage of jdisc_core.
- Add TODO to gc it later.

gjoranv committed:
Set up logging in ApplicationLoader again.

- If done in StandaloneMain, logging will not work in integration
  tests (using TestDriver) or test frameworks (e.g. application).

gjoranv committed:
Test vespa log instead of osgi log reader.

gjoranv committed:
Switch off jdisc ConsoleLogManager

- vespalog is now set up before loading the application.

gjoranv commented:
This should be good now, @baldersheim

baldersheim reviewed (approved):
Have you run any of the system tests that failed on last try ?
I think that is a good start to sanitise it prior to merge.

gjoranv commented:
Yes, I tried all tests that I could run locally, and I see no abnormal/unexpected logging now. I suspect the problem was the double call to initializeLog(). Must mention that I have a problem with the configserver shutdown in my container, so those tests never came to the log check, but the log looks fine. No curator/zk log errors/warnings now.
